operators/src/source/gdal_source.rs

```diff
@@ -0,0 +1,456 @@
+        assert_eq!(vres[34], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (5, 4).into(), (5000, 4000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[35], (TimeInterval::new_unchecked(1.into(),2.into()), TileInformation::new(global_size_in_tiles.into(), (5, 5).into(), (5000, 5000).into(), tile_size_in_pixels.into())));
+
+        assert_eq!(vres[36], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (0, 0).into(), (0, 0).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+1], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (0, 1).into(), (0, 1000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+6], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (1, 0).into(), (1000, 0).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+7], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (1, 1).into(), (1000, 1000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+34], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (5, 4).into(), (5000, 4000).into(), tile_size_in_pixels.into())));
+        assert_eq!(vres[36+35], (TimeInterval::new_unchecked(2.into(),3.into()), TileInformation::new(global_size_in_tiles.into(), (5, 5).into(), (5000, 5000).into(), tile_size_in_pixels.into())));
+    }
+    #[test]
+    fn test_load_tile_data() {
+        let global_size_in_pixels = (6000, 6000);
+        let tile_size_in_pixels = (1000, 1000);    
+        let global_size_in_tiles = (6, 6);
+
+
+        let grid_tile_provider = GdalSourceTileGridProvider {
+            global_pixel_size: global_size_in_pixels.into(),
+            tile_pixel_size: tile_size_in_pixels.into(),
+        };
+
+        let time_interval_provider = vec![TimeInterval::new_unchecked(0.into(),1.into())];
+        
+        let gdal_params = GdalSourceParameters{
+            base_path: "../operators/test-data/raster/srtm_38_03".into(),
+            file_name_with_time_placeholder: "srtm_38_03.tif".into(),
+            time_format: "".into(),
+            tick: None,
+            channel: None,
+        };
+
+        let gdal_source = GdalSource {
+            time_interval_provider,
+            grid_tile_provider,
+            gdal_params
+        };
+
+        let tile_information = TileInformation::new(global_size_in_tiles.into(), (0, 0).into(), (0, 0).into(), tile_size_in_pixels.into());
+        let time_interval = TimeInterval::new_unchecked(0.into(),1.into());
+
+        let x = gdal_source.load_tile_data::<f32>(&time_interval, &tile_information).unwrap();
+        
+        assert_eq!(x.tile, tile_information);
+        assert_eq!(x.time, time_interval);
```

Comment:
I am unsure about this one. You have to specify the sizes for the `GdalSource` constructor and then you check that they are still the same values? Shouldn't some values come from the source opening the data set? Or am I misunderstanding the test case?

Reply:
The Tile handling is more complex then i thought. So i went with an easy design. In this case the tile information is provided, however, the combination of time and tile is generated. This is tested here.